src/measure_theory/bochner_integration.lean

```diff
@@ -711,6 +711,132 @@ end simple_func_integral
 
 end simple_func
 
+open simple_func
+
+variables [normed_space ℝ β] [normed_space ℝ γ] [complete_space β]
+
+local notation `to_l1` := coe_to_l1 α β ℝ
+local notation `Integral` := @integral_clm α _ β _ _ _
+local attribute [instance] simple_func.normed_group simple_func.normed_space
+
+open continuous_linear_map
+
+/-- The Bochner integral in l1 space as a continuous linear map. -/
+def integral_clm : (α →₁ β) →L[ℝ] β :=
+  integral_clm.extend to_l1 simple_func.dense_range simple_func.uniform_inducing
+
+/-- The Bochner integral in l1 space -/
+def integral (f : α →₁ β) : β := (integral_clm).to_fun f
+
+variables (α β)
+@[simp] lemma integral_zero : integral (0 : α →₁ β) = 0 :=
+map_zero integral_clm
+variables {α β}
+
+lemma integral_add (f g : α →₁ β) : integral (f + g) = integral f + integral g :=
+map_add integral_clm f g
+
+lemma integral_neg (f : α →₁ β) : integral (-f) = - integral f :=
+map_neg integral_clm f
+
+lemma integral_sub (f g : α →₁ β) : integral (f - g) = integral f - integral g :=
+map_sub integral_clm f g
+
+lemma integral_smul (r : ℝ) (f : α →₁ β) : integral (r • f) = r • integral f :=
+map_smul r integral_clm f
+
 end l1
 
+variables [normed_group β] [second_countable_topology β] [normed_space ℝ β] [complete_space β]
+          [normed_group γ] [second_countable_topology γ] [normed_space ℝ γ] [complete_space γ]
+
+/-- The Bochner integral -/
+def integral (f : α → β) : β :=
+if hf : measurable f ∧ integrable f
+then (l1.of_fun f hf.1 hf.2).integral
+else 0
+
```

Comment:
since `integral` is irreducible at the end of this file, it will be useful to have a lemma recalling its definition to unfold it when necessary, i.e.,
```lemma integral_eq (f : α → β) :
  integral f = if hf : measurable f ...
```
and also lemmas `integral_zero_of_non_measurable` and `integral_zero_of_non_integrable`.